src/libsyntax/feature_gate.rs

```diff
@@ -385,6 +386,13 @@ impl<'a> Context<'a> {
                 return;
             }
         }
+        for &(ref n, ref ty) in self.plugin_attributes.iter() {
+            if &*n == name {
+                // Plugins can't gate attributes, so we don't check for it
```

Comment:
Above, not below

https://github.com/rust-lang/rust/pull/25168/files#diff-8fe611e80215042c8d89714f4dd6f8baR382